File: 0002/0002.cc
#include <iostream>
#include <gmpxx.h>

bool isPrime(mpz_class n)
{
	mpz_class m,o,p;
	for(m = 2; m < sqrt(n); m++)
		if(n % m == 0)
			return false;
	return true;
}

int main()
{
	for(mpz_class n = 2 ;; n++)
		if(isPrime(n))
			std::cout << n << "\n";
}

File: 0002/0002.hs
theSquirts :: Integer -> Integer;
theSquirts k
  | k < 2^29-1 = floor $ sqrt $ fromIntegral k
  | otherwise  = read $ take (length b `div` 2 + 1) b
    where b = show k

isPrime :: Integer -> Bool;
isPrime = (== []) . factors
  where factors k = [l | l <- [2..theSquirts k], mod k l == 0]

main :: IO [()];
main = mapM print $ filter isPrime [2..];

